api/client/src/intTest/java/org/projectnessie/client/auth/oauth2/ITOAuth2Client.java

```diff
@@ -44,27 +44,50 @@
 import org.keycloak.representations.idm.UserRepresentation;
 import org.keycloak.representations.idm.authorization.PolicyEnforcementMode;
 import org.keycloak.representations.idm.authorization.ResourceServerRepresentation;
-import org.projectnessie.client.auth.BasicAuthenticationProvider;
+import org.projectnessie.client.api.NessieApi;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.api.NessieApiV2;
 import org.projectnessie.client.http.HttpAuthentication;
-import org.projectnessie.client.http.HttpClient;
-import org.projectnessie.client.http.HttpClientException;
-import org.projectnessie.client.http.HttpRequest;
-import org.projectnessie.client.http.HttpResponse;
+import org.projectnessie.client.http.HttpClientBuilder;
 import org.projectnessie.client.http.Status;
+import org.projectnessie.client.rest.NessieNotAuthorizedException;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
 @Testcontainers
 @ExtendWith(SoftAssertionsExtension.class)
 public class ITOAuth2Client {
 
+  static Network network = Network.newNetwork();
+
   @Container
   private static final KeycloakContainer KEYCLOAK =
-      new KeycloakContainer().withFeaturesEnabled("preview", "token-exchange");
+      new KeycloakContainer()
+          .withNetwork(network)
+          .withNetworkAliases("keycloak")
+          .withEnv("KC_HOSTNAME_URL", "http://keycloak:8080")
+          .withFeaturesEnabled("preview", "token-exchange");
+
+  @Container
+  @SuppressWarnings("resource")
+  private static final GenericContainer<?> NESSIE =
+      new GenericContainer<>("ghcr.io/projectnessie/nessie:latest")
```

Comment:
Why not run the end-to-end test under the quarkus server module as another sub-class of `BaseClientAuthTest`?

Reply:
Why not... but I'm not sure as we are testing here some internals of `OAuth2Client` and besides, `OAuth2Client` is package-private. It might not be so easy to move this to the server module. But I can look into adding another test there, maybe something less involved. Would that work for you?

Reply:
I mean we could probably keep the majority of tests for `OAuth2Client` here and validate against Keycloak. Then, just have a basic end-to-end test via Quarkus under the quarkus module so that we would not have to depend on Nessie docker images.

Reply:
Yea - better some higher-level tests in `:nessie-quarkus`. The `latest` tag refers to the latest Nessie release, not the current state of the code, so it's tricky. BTW: we have the apprunner-plugin to run a Nessie using the in-tree Nessie code. But pulling that into `:nessie-client` would implicitly pull `:nessie-quarkus` for ITs into `:nessie-client`, not strictly a cyclic dependency, but not a "nice" depeendency cycle.

TL;DR it's probably better to have high-level in `:nessie-quarkus`.

Reply:
Alright, I added 2 classes in nessie-quarkus:

* `ITOAuth2Authentication` uses a true Keycloak server and a newly-introduced test resource for Keycloak.
* `TestOAuth2Authentication` uses Wiremock and `OidcWiremockTestResource`, just like the Bearer tests.

Let me know what you think. If you prefer that I rollback the changes in `ITOAuthClient`, no problem. Now that we have tests in nessie-quarkus, it's not so much of a problem if we don't bring Nessie in in `ITOAuthClient`

Reply:
Yea, I'd prefer to remove Nessie from the IT in `:nessie-client`